### PR TITLE
Replace spacesInsteadTabs with indentUsing

### PR DESCRIPTION
**Pull request by nth-chile**
Replaces the boolean option to use 4 spaces instead of tabs with a dropdown menu allowing you to select Tab, 2 Spaces, or 4 Spaces.

Please let me know if you agree with this change.

### CODE REVIEW

**Comment by glushchenko** [comment]
I cant accept this pull request:

1) Layout in settings is broken (wrong margins at bottom, not centred items)
2) Auto indent not applied when new return pressed. 
3)The project is not building due to errors in Arabic localization.

**Comment by nth-chile** [comment]
Ok, working on it

**Comment by glushchenko** [comment]
Thanks @nth-chile, i will be grateful.

**Comment by nth-chile** [comment]
I've identified the places where I still need to add the case for two spaces (for example, creating newlines), but I'm stuck until I figure out why `print("test")` and `NSLog("test")` don't write to the console!

**Comment by nth-chile** [comment]
Ok, added newlines and untabbing (though I'm not sure how to untab in the app. Shouldn't it be shift + tab?). I have a few questions, and potentially found a typo/bug in the master branch.

First, here is the potential typo.


```
FSNotes/Helpers/TextFormatter:476

        if string.starts(with: "\t") {
            padding = 1
        } else if string.starts(with: "   ") { // This string is three spaces. Should it be 4?
            padding = 4
        }
```

Once I know why that is three spaces and not 4, then I will add the condition for 2 space tab setting there.

I didn't want to make any assumptions about how code blocks work. Could you tell me if any of the following lines need to be adjusted for this PR?


```
FSNotes/Helpers/NotesTextProcessor:960

if parString.starts(with: "    ") || parString.starts(with: "\t") {
    return
}
```


```
FSNotes/Business/Note:1142

if par.starts(with: "    ") || par.starts(with: "\t") {
    return
}
```


```
FSNotesCore/Core macOS/Extensions/NSTextStorage+:85

...
 value.starts(with: "    ")
                || value.starts(with: "\t")
                || value.starts(with: "* ")
                || value.starts(with: "- ")
                || value.starts(with: "+ ")
                || value.starts(with: "> ")
                || self.getNumberListPrefix(paragraph: value) != nil
...
```


```
FSNotesCore/Core iOS/Extensions/NSTextStorage+:85

...
 value.starts(with: "    ")
                || value.starts(with: "\t")
                || value.starts(with: "* ")
                || value.starts(with: "- ")
                || value.starts(with: "+ ")
                || value.starts(with: "> ")
                || self.getNumberListPrefix(paragraph: value) != nil
...
```


```
FSNotesCore/Shared/CodeTextProcessor:215

    public func isCodeBlockParagraph(_ paragraph: String) -> Bool {
        if paragraph.starts(with: "\t") || paragraph.starts(with: "    ") {
            return true
        }

        return false
    }
```



```
FSNotesCore/Shared/CodeTextProcessor:142

            if let value = value, (value.starts(with: "    ") || value.starts(with: "\t")), self.textStorage.attributedSubstring(from: range).hasTodoAttribute() {
                return
            }
```


```
FSNotesCore/Shared/CodeTextProcessor:127

    private func isCodeBlock(_ attributedString: NSAttributedString) -> Bool {
        if attributedString.string.starts(with: "\t") || attributedString.string.starts(with: "    ") {
            return true
        }

        return false
    }
```

Sorry for all the questions, I just want to be thorough and not introduce bugs!

**Comment by glushchenko** [comment]
Hi, @nth-chile. Sorry for delayed response. Preferences layout looks like it broken. RTL not supported and elements not aligned like other elements.

What do you suggest to do with markdown, two spaces doesn't seem to work either?

**Comment by glushchenko** [comment]
@nth-chile please test last changes, seems this final version.

File: FSNotes/Helpers/TextFormatter.swift [hunk 1]
```diff
@@ -499,7 +499,7 @@ public class TextFormatter {
                     firstCharsToDrop = 2
                 }
                 
-                if line.starts(with: "    ") {
+                if UserDefaultsManagement.indentUsing == 0x02 && line.starts(with: "    ") {
```
Comment (glushchenko) [inline FSNotes/Helpers/TextFormatter.swift:502]: I dont think that is good case. When Tab selected, untab for spaces should works too.

**Comment by nth-chile** [comment]
@glushchenko Thanks! How did you do all the languages? Do you just use Google translate? 

I did find one bug:

- Set indentUsing to 2 Spaces
- Indent twice, type something, press enter for newline
- Untab creates additional newline

I added a commit to fix this, let me know if that looks ok.

In my previous comment I listed all of the spots where I found "/t" and "    " in the code. If you think everything is ok, then the PR tests well for me

File: FSNotes/Helpers/TextFormatter.swift (hunk 1 above)
Comment (nth-chile) [reply FSNotes/Helpers/TextFormatter.swift:502]: I agree. I ran into an issue with two indents of two spaces "    " that I described below. Was trying to fix that here

**Comment by nth-chile** [comment]
https://user-images.githubusercontent.com/22358233/145719916-653a52c8-7ca5-4036-b8fb-31366d796338.mov

**Comment by glushchenko** [comment]
Please checkout last changes

**Comment by nth-chile** [comment]
Looks good to me. Thanks!

**Comment by glushchenko** [comment]
Released here:

https://github.com/glushchenko/fsnotes/releases/download/5.2.0/FSNotes_5.2.0.zip

**Pull request merged**